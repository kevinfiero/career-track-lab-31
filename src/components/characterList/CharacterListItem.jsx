import React from 'react';
import { Link } from 'react-router-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import PropTypes from 'prop-types';
import './CharacterListItem.css';

const CharacterListItem = ({ image, name, id }) => (
  <Router>
    <Link to={`./${id}`}>
      <figure>
        <img src={image} />
        <figcaption>
          <h1>{name}</h1>
        </figcaption>
      </figure>
    </Link>
  </Router>

);

CharacterListItem.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default CharacterListItem;
