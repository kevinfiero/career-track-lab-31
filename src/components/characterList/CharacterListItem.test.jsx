import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import CharacterListItem from './CharacterListItem';

describe('CharacterListItem component', () => {
  afterEach(() => cleanup());
  it('renders CharacterListItem', () => {
    const { asFragment } = render(
      <Router>
        <CharacterListItem 
          image="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
          name="Arnold Shortman"
          id="5da237699734fdcb7bef8f51"
        />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
