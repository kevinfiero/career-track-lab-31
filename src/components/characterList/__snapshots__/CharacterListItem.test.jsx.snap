// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterListItem component renders CharacterListItem 1`] = `
<DocumentFragment>
  <a
    href="/5da237699734fdcb7bef8f51"
  >
    <figure>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
      />
      <figcaption>
        <h1>
          Arnold Shortman
        </h1>
      </figcaption>
    </figure>
  </a>
</DocumentFragment>
`;
